cowbird/monitoring/monitoring.py

```diff
@@ -1,177 +1,103 @@
-            # create event neither
-            if self.__recursive or \
-                    os.path.dirname(event.dest_path) == \
-                    os.path.dirname(self.__src_path):
-                self.__callback.on_created(event.dest_path)
-
-    def on_created(self, event):
-        # type: (Union[DirCreatedEvent, FileCreatedEvent]) -> None
-        """
-        Called when a file or directory is created.
-
-        @param event: Event representing file/directory creation.
-        """
-        self.__callback.on_created(event.src_path)
-
-    def on_deleted(self, event):
-        # type: (Union[DirDeletedEvent, FileDeletedEvent]) -> None
-        """
-        Called when a file or directory is deleted.
-
-        @param event: Event representing file/directory deletion.
-        """
-        self.__callback.on_deleted(event.src_path)
-
-    def on_modified(self, event):
-        # type: (Union[DirModifiedEvent, FileModifiedEvent]) -> None
-        """
-        Called when a file or directory is modified.
-
-        @param event: Event representing file/directory modification.
-        """
-        self.__callback.on_modified(event.src_path)
+LOGGER = get_logger(__name__)
 
 
-@six.add_metaclass(SingletonMeta)
-class Monitoring:
+class Monitoring(metaclass=SingletonMeta):
     """
     Class handling file system monitoring and registering listeners.
 
     .. todo:: At some point we will need a consistency function that goes through all monitored folder and make sure
               that monitoring services are up to date.
     """
 
-    def __init__(self):
+    def __init__(self, config=None):
+        # type: (AnySettingsContainer) -> None
+        """
+        Initialize the monitoring instance from configured application settings.
+
+        @param config: AnySettingsContainer object from which the db can be retrieved.
+                       The default value of None is only there to disable pylint E1120. The singleton instance
+                       must be initialized with a proper config and after that the init function should not be hit.
+        """
+        if not config:
+            raise Exception("Without proper application settings, the Monitoring class cannot obtains a proper database"
+                            " store.")
         self.monitors = defaultdict(lambda: {})
+        self.store = get_db(config).get_store(MonitoringStore)
 
     def start(self):
         """
-        .. todo:: Load and start monitors from the BD
+        Load existing monitors and start the monitoring.
         """
+        monitors = self.store.list_monitors()
+        for mon in monitors:
+            self.monitors[mon.path][mon.callback] = mon
+            mon.start()
 
     def register(self, path, recursive, cb_monitor):
-        # type: (str, bool, FSMonitor) -> None
+        # type: (str, bool, Union[FSMonitor, Type[FSMonitor], str]) -> Monitor
         """
         Register a monitor for a specific path and start it.
```

Comment:
add note about already registered monitor is returned instead of re-register/conflict-error